File: 977-SquaresOfSortedArray/twoPointers.java
class Solution {
    public int[] sortedSquares(int[] A) {
        int len = A.length;
        int i = 0, j = len - 1;
        //From the start and end, which can solve the negative numbers in the front
        int[] res = new int[len];
        for (int k = len - 1; k >= 0; k--) {
            int sq1 = A[i] * A[i];
            int sq2 = A[j] * A[j];
            if (sq1 < sq2) {
                res[k] = sq2;
                j--;
            }
            else {
                res[k] = sq1;
                i++;
            }
        }
        return res;
    }
}

File: 167-TwoSumII-InputSorted/twoPointers.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int len = numbers.size();
        int start = 0, end = len - 1;
        vector<int> res(2);
        while (start < end) {
            if (numbers[start] + numbers[end] < target) start++;
            else if (numbers[start] + numbers[end] > target) end--;
            else {
                res[0] = start + 1;
                res[1] = end + 1;
                break;
            }
        }
        return res;
    }
};